.github/workflows/performance-benchmarks-kurobako.yml

```diff
@@ -0,0 +1,144 @@
+        required: false
+        default: 'NopPruner'
+      pruner-kwargs-list:
+        description: 'Pruner Arguments List: A list of pruner keyword arguments. Should be a whitespace-separated list of json format dictionaries.'
+        required: false
+        default: '{}'
+      extra-requirements:
+        description: 'Extra Requirements: A list of extra requirements to run the benchmark. Should be a whitespace-separated list of packaged which can be installed by `pip`.'
+        required: false
+        default: ''
+      n-runs:
+        description: 'Number of Studies'
+        required: false
+        default: '10'
+
+
+jobs:
+  performance-benchmarks-kurobako:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup Python3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+
+    - name: Install gnuplot
+      run: |
+        sudo apt update
+        sudo apt -y install gnuplot
+
+    - name: Setup cache
+      uses: actions/cache@v2
+      env:
+        # Caches them under a common name so that they can be used by other performance benchmark.
+        cache-name: performance-benchmarks
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-3.9-${{ env.cache-name }}-${{ hashFiles('**/setup.py') }}-v1
+        restore-keys: |
+          ${{ runner.os }}-3.9-${{ env.cache-name }}-${{ hashFiles('**/setup.py') }}
+
+    - name: Install Python libralies
+      run: |
+        python -m pip install --upgrade pip
+        pip install --progress-bar off -U setuptools
+
+        # Install minimal dependencies and confirm that `import optuna` is successful.
+        pip install --progress-bar off .
+        python -c 'import optuna'
+        optuna --version
+
+        pip install --progress-bar off .[benchmark]
+
+        pip install --progress-bar off kurobako ${{ github.event.inputs.extra-requirements }}
+
+    - name: Cache kurobako CLI
+      id: cache-kurobako
+      uses: actions/cache@v2
+      with:
+        path: ./kurobako
+        key: kurobako-0-2-9
+
+    - name: Install kurobako CLI
+      if: steps.cache-kurobako.outputs.cache-hit != 'true'
+      run: |
+        curl -L https://github.com/optuna/kurobako/releases/download/0.2.9/kurobako-0.2.9.linux-amd64 -o kurobako
+
+        chmod +x kurobako
+        ./kurobako -h
+
+    - name: Cache hpobench dataset
+      id: cache-hpobench-dataset
+      uses: actions/cache@v2
+      with:
+        path: ./fcnet_tabular_benchmarks
+        key: hpobench-dataset
+
+    - name: Download hpobench dataset
+      if: steps.cache-hpobench-dataset.outputs.cache-hit != 'true'
+      run: |
+        wget http://ml4aad.org/wp-content/uploads/2019/01/fcnet_tabular_benchmarks.tar.gz
+        tar xf fcnet_tabular_benchmarks.tar.gz
```

Comment:
Just checking but can caching ever become an issue with such large files?

Reply:
According to [here](https://github.com/actions/cache#cache-limits), the cache limit for GitHub Actions is 10 GB, and old or recently unused data is automatically deleted. We don't need to manage the cache, which means that no particular problem will occur, I think.